managed/src/main/java/com/yugabyte/yw/controllers/CustomerConfigController.java

```diff
@@ -61,4 +60,36 @@ public Result delete(UUID customerUUID, UUID configUUID) {
   public Result list(UUID customerUUID) {
     return ApiResponse.success(CustomerConfig.getAll(customerUUID));
   }
+
+  public Result edit(UUID customerUUID, UUID configUUID) {
+    JsonNode formData =  request().body().asJson();
+    ObjectNode errorJson = configValidator.validateFormData(formData);
+    if (errorJson.size() > 0) {
+      return ApiResponse.error(BAD_REQUEST, errorJson);
+    }
+
+    errorJson = configValidator.validateDataContent(formData);
+    if (errorJson.size() > 0) {
+      return ApiResponse.error(BAD_REQUEST, errorJson);
+    }
+    CustomerConfig customerConfig = CustomerConfig.get(customerUUID, configUUID);
+    if (customerConfig == null) {
+      return ApiResponse.error(BAD_REQUEST, "Invalid configUUID: " + configUUID);
+    }
+    CustomerConfig config = CustomerConfig.get(configUUID);
+    JsonNode data = Json.toJson(formData.get("data"));
+    for (Iterator<String> it = data.fieldNames(); it.hasNext(); ) {
+      String key = it.next();
+      if ((key.contains("KEY") || key.contains("SECRET") || key.contains("CREDENTIALS"))
```

Comment:
I know this code is used in some other places but I don't like it overall. There are at least two examples when it works badly:
1) When UI gets passwords they are masked as 'ab****yz' - two first and two last characters are unmasked. This gives information about the password length and doesn't mask well short passwords (apple -> 'ap*le', 'password' -> 'pa****rd').
2) With your code it is impossible to set passwords with two asterisks inside.

Other comments:
We have the same condition in another class/function - CustomerConfig.java :: getData, I propose to create a static function something like CustomerConfig.isSensitiveKey(...) which returns `true` if the field contains sensitive data or `false` otherwise. And then to use it.

Reply:
I totally agree to 1 and 2. But what is the suitable condition to check whether it is sensitive data or not?  Lenght should be more than 4 characters and some other condition?
Also, I do not think these fields won't be having short keys as it is provider credentials or keys..But totally agree to your poin tbh.

Reply:
@SergeyPotachev 
@Arnav15 suggested 
```if (maskConfig(config.getData()) != data){
      config.data = Json.toJson(data);
    }```
 if the incoming request data and the masked DB data is not same then update the data field.
 
 Also, there 3 fields are not passwords and it will be long ( atleast for these 3 cases (SECRET, KEY , CREDENTIAL ) )
 
 with the change now user can update the fields with ** too.
 
 LMK if any other changes needed here.

Reply:
@SergeyPotachev 
@Arnav15 suggested 
```
  if (maskConfig(config.getData()) != data){
      config.data = Json.toJson(data);
    }
```
If the incoming data for edit is not equal to the existing DB's masked data, then update the data in DB.
LMK if you need any other validations here.